packages/gcc10.rb

```diff
@@ -69,128 +76,338 @@ def self.build
       when 'x86_64'
-        system "../configure  #{CREW_OPTIONS} \
-        --enable-checking=release \
-        --enable-lto \
-        --disable-multilib \
-        --enable-threads=posix \
-        --disable-werror \
-        --disable-libmpx \
-        --enable-static \
-        --enable-shared \
-        --with-pic \
-        --program-suffix=-#{gcc_version} \
-        --with-arch-64=x86-64 \
-        --with-build-config=bootstrap-lto-lean"
+        system "env CFLAGS=#{@cflags} CXXFLAGS=#{@cxxflags}  \
+          LD_LIBRARY_PATH=#{CREW_LIB_PREFIX} \
+          LIBRARY_PATH=#{CREW_LIB_PREFIX} PATH=#{@path} \
+          ../configure #{CREW_OPTIONS} \
+          --disable-libmpx \
+          --disable-libssp \
+          --disable-multilib \
+          --disable-werror \
+          --enable-cet=auto \
+          --enable-checking=release \
+          --enable-clocale=gnu \
+          --enable-default-pie \
+          --enable-default-ssp \
+          --enable-gnu-indirect-function \
+          --enable-gnu-unique-object \
+          --enable-host-shared \
+          --enable-languages=#{@languages}  \
+          --enable-lto \
+          --enable-plugin \
+          --enable-shared \
+          --enable-static \
+          --enable-threads=posix \
+          --program-suffix=-#{gcc_version} \
+          --with-arch-64=x86-64 \
+          --with-build-config=bootstrap-lto-lean \
+          --with-isl \
+          --with-pic \
+          --with-system-zlib"
       when 'i686'
-        system "../configure  #{CREW_OPTIONS} \
-        --enable-checking=release \
-        --enable-lto \
-        --disable-multilib \
-        --enable-threads=posix \
-        --disable-werror \
-        --disable-libmpx \
-        --enable-static \
-        --enable-shared \
-        --with-pic \
-        --program-suffix=-#{gcc_version} \
-        --with-arch-32=#{ARCH} \
-        --with-build-config=bootstrap-lto-lean"
+        system "env CFLAGS=#{@cflags} CXXFLAGS=#{@cxxflags} \
+          LD_LIBRARY_PATH=#{CREW_LIB_PREFIX} \
+          LIBRARY_PATH=#{CREW_LIB_PREFIX} PATH=#{@path} \
+          ../configure  #{CREW_OPTIONS} \
+          --disable-libmpx \
+          --disable-libssp \
+          --disable-multilib \
+          --disable-werror \
+          --enable-cet=auto \
+          --enable-checking=release \
+          --enable-clocale=gnu \
+          --enable-default-pie \
+          --enable-default-ssp \
+          --enable-gnu-indirect-function \
+          --enable-gnu-unique-object \
+          --enable-host-shared \
+          --enable-languages=#{@languages}  \
+          --enable-lto \
+          --enable-plugin \
+          --enable-shared \
+          --enable-static \
+          --enable-threads=posix \
+          --program-suffix=-#{gcc_version} \
+          --with-arch-32=#{ARCH} \
+          --with-build-config=bootstrap-lto-lean \
+          --with-isl \
+          --with-pic \
+          --with-system-zlib"
```

Comment:
I'm wondering instead of including all the same configure options for each architecture if it's possible to have one set of options for every architecture with variables?

Reply:
Just like llvm, sure...

That's a lot easier to do post hoc once we know that compiles work...

Reply:
Collapsed all arch versions into one section.